sstables/kl/reader_impl.hh

```diff
@@ -215,7 +210,12 @@ private:
             return row_consumer::proceed::yes;
         }
```

Comment:
Probably we can move the testing hack outside this code and rewrite it as an interposer input_stream that returns 1-byte fragments (or even randomly sized fragments). This can allow it to live outside a #ifdef and participate in regular testing. I'm not suggesting to do this as part of this series.

Reply:
Don't we actually have something like that in utils/limiting_data_source.cc?

Reply:
Yes